pyg_lib/csrc/ops/matmul.cpp

```diff
@@ -19,6 +22,60 @@ std::vector<at::Tensor> _grouped_matmul(const std::vector<at::Tensor>& input,
   return op.call(input, other);
 }
 
+std::vector<at::Tensor> concat(std::vector<at::Tensor> t1,
+                               std::vector<at::Tensor> t2) {
+  for (size_t i = 0; i < t2.size(); ++i) {
+    t1.push_back(t2[i]);
+  }
+  return t1;
+}
+
+auto get_input(const std::vector<at::Tensor>& t, int split_index) {
+  std::vector<at::Tensor> t1(t.begin(), t.begin() + split_index);
+  return t1;
+}
+
+auto get_other(const std::vector<at::Tensor>& t, int split_index) {
+  std::vector<at::Tensor> t2(t.begin() + split_index, t.end());
+  return t2;
+}
+
+class GroupedMatmul : public torch::autograd::Function<GroupedMatmul> {
+  // TODO (matthias) Add TensorArg definitions.
+ public:
+  static variable_list forward(AutogradContext* ctx,
+                               std::vector<Variable> input,
+                               std::vector<Variable> other) {
+    auto out = _grouped_matmul(input, other);
+    variable_list input_and_other = concat(input, other);
+    ctx->save_for_backward(input_and_other);
+    ctx->saved_data["input_len"] = (int)input.size();
+    return out;
+  }
+
+  static variable_list backward(AutogradContext* ctx, variable_list grad_outs) {
+    auto input_and_other = ctx->get_saved_variables();
+    int input_len = ctx->saved_data["input_len"].toInt();
+    auto input = get_input(input_and_other, input_len);
```

Comment:
Let's just do this inline? I don't think we need to specify `get_input` and `get_other` here.